packages/web-components/src/components/va-alert-sign-in/va-alert-sign-in.tsx

```diff
@@ -0,0 +1,312 @@
+import {
+  Component,
+  Element,
+  // Event,
+  // EventEmitter,
+  Host,
+  Prop,
+  h,
+} from '@stencil/core';
+import classnames from 'classnames';
+import { VariantNames } from './VariantNames';
+import { getHeaderLevel } from '../../utils/utils';
+
+/**
+ * @componentName Alert - Sign In
+ * @maturityCategory caution
+ * @maturityLevel candidate
+ */
+@Component({
+  tag: 'va-alert-sign-in',
+  styleUrl: 'va-alert-sign-in.scss',
+  shadow: true,
+})
+export class VaAlertSignIn {
+  @Element() el!: any;
+
+  /**
+   * Determines the text content and border/background color. Must be one of "signInRequired", "signInOptional", "signInEither", "verifyIdMe", or "verifyLoginGov".
+   */
+  @Prop() variant?: string = VariantNames.signInRequired;
+
+  /**
+   * Header level for button wrapper. Must be between 1 and 6
+   */
+  @Prop() headingLevel?: number = 2;
+
+  /**
+   * If `true`, doesn't fire the CustomEvent which can be used for analytics tracking.
+   */
+  @Prop() disableAnalytics?: boolean = true;
+
+  /**
+   * If `true`, the alert will be visible.
+   */
+  @Prop() visible?: boolean = true;
+
+  /**
+   * For the 'optional' variant, how long the respondent has to submit their form
+   */
+  @Prop() timeLimit?: string = '15 minutes';
+
+  /**
+   * For the 'optional' variant, the link to the form to complete without signing in
+   */
+  @Prop() noSignInLink?: string;
+
+  /**
+   * The event used to track usage of the component. This is emitted when an
+   * anchor link is clicked and disableAnalytics is not true.
+   */
+  // @Event({
+  //   eventName: 'component-library-analytics',
+  //   composed: true,
+  //   bubbles: true,
+  // })
+  // componentLibraryAnalytics: EventEmitter;
+
+  // private handleAlertBodyClick(e: MouseEvent): void {
+  //   if (!this.disableAnalytics) {
+  //     const target = e.target as HTMLElement;
+
+  //     // If it's a link being clicked, dispatch an analytics event
+  //     if (target?.tagName === 'VA-LINK') {
+  //       const innerText = target.shadowRoot.querySelector('a').innerText;
+  //       const detail = {
+  //         componentName: 'va-alert-sign-in',
+  //         action: 'linkClick',
+  //         details: {
+  //           clickLabel: innerText,
+  //           variant: this.variant,
+  //         },
+  //       };
+  //       this.componentLibraryAnalytics.emit(detail);
+  //     }
+  //   }
```

Comment:
Can we clean up all of the commented out code?

Reply:
Since we'd need all of this code once analytics is worked out in a week or two, it's my preference to just leave it here.

Reply:
@Andrew565 Can we link to that analytics ticket here then? Is it in a sprint yet?